diff --git a/mqemitter.d.ts b/mqemitter.d.ts
index 874de6c..0e7a986 100644
--- a/mqemitter.d.ts
+++ b/mqemitter.d.ts
@@ -15,7 +15,7 @@ declare namespace mqemitter {
     current: number
     concurrent: number
     on(topic: string, listener: (message: Message, done: () => void) => void, callback?: () => void): this
-    emit(message: Message, callback?: (error?: Error) => void): void
+    once(topic: string, listener: (message: Message, done: () => void) => void, callback?: () => void): this
     removeListener(topic: string, listener: (message: Message, done: () => void) => void, callback?: () => void): void
     close(callback: () => void): void
   }
diff --git a/mqemitter.js b/mqemitter.js
index 7d556e8..5c8d591 100644
--- a/mqemitter.js
+++ b/mqemitter.js
@@ -95,24 +95,21 @@ MQEmitter.prototype.removeAllListeners = function removeListener (topic, done) {
   return this
 }
 
-MQEmitter.prototype.emit = function emit (message, cb) {
-  assert(message)
+MQEmitter.prototype.once = function once (topic, notify, done) {
+  assert(topic)
+  assert(notify)
 
-  cb = cb || noop
+  const that = this
 
-  if (this.closed) {
-    return cb(new Error('mqemitter is closed'))
+  function wrapper (message, cb) {
+    that._matcher.remove(topic, wrapper)
+    notify(message, cb)
   }
 
-  if (this.concurrency > 0 && this.current >= this.concurrency) {
-    this._messageQueue.push(message)
-    this._messageCallbacks.push(cb)
-    if (!this._doing) {
-      process.emitWarning('MqEmitter leak detected', { detail: 'For more info check: https://github.com/mcollina/mqemitter/pull/94' })
-      this._released()
-    }
-  } else {
-    this._do(message, cb)
+  this._matcher.add(topic, wrapper)
+
+  if (done) {
+    setImmediate(done)
   }
 
   return this
