diff --git a/mqemitter.d.ts b/mqemitter.d.ts
index 874de6c..ab0f9fe 100644
--- a/mqemitter.d.ts
+++ b/mqemitter.d.ts
@@ -15,7 +15,6 @@ declare namespace mqemitter {
     current: number
     concurrent: number
     on(topic: string, listener: (message: Message, done: () => void) => void, callback?: () => void): this
-    emit(message: Message, callback?: (error?: Error) => void): void
     removeListener(topic: string, listener: (message: Message, done: () => void) => void, callback?: () => void): void
     close(callback: () => void): void
   }
diff --git a/mqemitter.js b/mqemitter.js
index 7d556e8..4861801 100644
--- a/mqemitter.js
+++ b/mqemitter.js
@@ -95,22 +95,6 @@ MQEmitter.prototype.removeAllListeners = function removeListener (topic, done) {
   return this
 }
 
-MQEmitter.prototype.emit = function emit (message, cb) {
-  assert(message)
-
-  cb = cb || noop
-
-  if (this.closed) {
-    return cb(new Error('mqemitter is closed'))
-  }
-
-  if (this.concurrency > 0 && this.current >= this.concurrency) {
-    this._messageQueue.push(message)
-    this._messageCallbacks.push(cb)
-    if (!this._doing) {
-      process.emitWarning('MqEmitter leak detected', { detail: 'For more info check: https://github.com/mcollina/mqemitter/pull/94' })
-      this._released()
-    }
   } else {
     this._do(message, cb)
   }
